test/__init__.py

```diff
@@ -84,25 +90,25 @@ def has_alpn(ctx_cls=None):
 RESOLVES_LOCALHOST_FQDN = _can_resolve("localhost.")
 
 
-def clear_warnings(cls=HTTPWarning):
+def clear_warnings(cls: type = HTTPWarning) -> None:
     new_filters = []
-    for f in warnings.filters:
+    for f in warnings.filters:  # type: ignore
         if issubclass(f[2], cls):
             continue
         new_filters.append(f)
-    warnings.filters[:] = new_filters
+    warnings.filters[:] = new_filters  # type: ignore
 
 
-def setUp():
+def setUp() -> None:
     clear_warnings()
     warnings.simplefilter("ignore", HTTPWarning)
 
 
-def notWindows(test):
+def notWindows(test: Callable[..., Any]) -> Callable[..., Any]:
```

Comment:
I think we want to make these decorators generic, so the type of callable they receive is the type of callable they return. Otherwise, the underlying test type becomes Any and (I think) isn't checked.